<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Export_Video_Widget</class>
 <widget class="QWidget" name="Export_Video_Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>243</width>
    <height>130</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="title_sequence_groupbox">
     <property name="title">
      <string>Title Sequence</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <layout class="QVBoxLayout" name="title_layout">
      <item>
       <layout class="QHBoxLayout" name="title_frames_layout">
        <item>
         <widget class="QSpinBox" name="title_frames_spinbox">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>100</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="title_frames_label">
          <property name="text">
           <string>Title Frames</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="font_size_layout">
        <item>
         <widget class="QSpinBox" name="font_size_spinbox">
          <property name="minimum">
           <number>12</number>
          </property>
          <property name="maximum">
           <number>72</number>
          </property>
          <property name="value">
           <number>24</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="font_size_label">
          <property name="text">
           <string>Font Size</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="title_text_label">
        <property name="text">
         <string>Title Text:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTextEdit" name="title_text_edit">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>60</height>
         </size>
        </property>
        <property name="html">
         <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:7.8pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Subject 100&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;10 Hz stimulation&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="preview_label">
        <property name="text">
         <string>Preview:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="title_preview">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>100</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>100</height>
         </size>
        </property>
        <property name="styleSheet">
         <string>background-color: black; color: white; border: 1px solid gray;</string>
        </property>
        <property name="text">
         <string>Preview</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignCenter</set>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="media_source_groupbox">
     <property name="title">
      <string>Media Source</string>
     </property>
     <layout class="QVBoxLayout" name="media_source_layout">
      <item>
       <layout class="QHBoxLayout" name="media_widget_layout">
        <item>
         <widget class="QLabel" name="media_widget_label">
          <property name="text">
           <string>Media Widget:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="media_widget_combobox">
          <property name="toolTip">
           <string>Select which media widget's content to export</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="sequences_groupbox">
     <property name="title">
      <string>Video Sequences</string>
     </property>
     <layout class="QVBoxLayout" name="sequences_layout">
      <item>
       <widget class="QLabel" name="sequences_info_label">
        <property name="text">
         <string>Add multiple frame sequences to concatenate into one video. Leave empty to use single sequence mode.</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
        <property name="styleSheet">
         <string>color: gray; font-style: italic;</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="sequence_buttons_layout">
        <item>
         <widget class="QPushButton" name="add_sequence_button">
          <property name="text">
           <string>Add Sequence</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="remove_sequence_button">
          <property name="text">
           <string>Remove Selected</string>
          </property>
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="button_spacer">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTableWidget" name="sequences_table">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>150</height>
         </size>
        </property>
        <property name="alternatingRowColors">
         <bool>true</bool>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
        </property>
        <property name="gridStyle">
         <enum>Qt::PenStyle::SolidLine</enum>
        </property>
        <column>
         <property name="text">
          <string>Start Frame</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>End Frame</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Has Title</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Title Text</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Title Frames</string>
         </property>
        </column>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="mode_separator">
     <property name="text">
      <string>─── Single Sequence Mode (used when no sequences above) ───</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
     <property name="styleSheet">
      <string>color: gray; font-style: italic;</string>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QSpinBox" name="start_frame_spinbox"/>
     </item>
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Start Frame</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QSpinBox" name="end_frame_spinbox">
       <property name="minimum">
        <number>-1</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>End Frame</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="frame_rate_layout">
     <item>
      <widget class="QSpinBox" name="frame_rate_spinbox">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>120</number>
       </property>
       <property name="value">
        <number>30</number>
       </property>
       <property name="suffix">
        <string> fps</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="frame_rate_label">
       <property name="text">
        <string>Frame Rate</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QLabel" name="duration_estimate_label">
     <property name="text">
      <string>Estimated Duration: 0.0 seconds</string>
     </property>
     <property name="styleSheet">
      <string>color: blue; font-weight: bold;</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="output_settings_groupbox">
     <property name="title">
      <string>Output Settings</string>
     </property>
     <layout class="QVBoxLayout" name="output_settings_layout">
      <item>
       <widget class="QLineEdit" name="output_filename">
        <property name="text">
         <string>output.mp4</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="output_size_layout">
        <item>
         <widget class="QSpinBox" name="output_width_spinbox">
          <property name="minimum">
           <number>64</number>
          </property>
          <property name="maximum">
           <number>7680</number>
          </property>
          <property name="value">
           <number>1920</number>
          </property>
          <property name="suffix">
           <string> px</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="width_label">
          <property name="text">
           <string>Width</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="output_height_spinbox">
          <property name="minimum">
           <number>64</number>
          </property>
          <property name="maximum">
           <number>4320</number>
          </property>
          <property name="value">
           <number>1080</number>
          </property>
          <property name="suffix">
           <string> px</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="height_label">
          <property name="text">
           <string>Height</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="reset_to_media_size_button">
          <property name="text">
           <string>Reset to Media Size</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="export_video_button">
     <property name="text">
      <string>Export!</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="audio_output_groupbox">
     <property name="title">
      <string>Audio Output</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <layout class="QVBoxLayout" name="audio_layout">
      <item>
       <widget class="QLabel" name="audio_description_label">
        <property name="text">
         <string>Generate audio clicks from digital event series aligned with video timeline.</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
        <property name="styleSheet">
         <string>color: gray; font-style: italic;</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="audio_settings_layout">
        <item>
         <widget class="QSpinBox" name="audio_sample_rate_spinbox">
          <property name="minimum">
           <number>8000</number>
          </property>
          <property name="maximum">
           <number>96000</number>
          </property>
          <property name="value">
           <number>44100</number>
          </property>
          <property name="suffix">
           <string> Hz</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="audio_sample_rate_label">
          <property name="text">
           <string>Audio Sample Rate</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="click_duration_spinbox">
          <property name="minimum">
           <double>0.001000000000000</double>
          </property>
          <property name="maximum">
           <double>1.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.010000000000000</double>
          </property>
          <property name="value">
           <double>0.050000000000000</double>
          </property>
          <property name="suffix">
           <string> s</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="click_duration_label">
          <property name="text">
           <string>Click Duration</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="audio_sources_label">
        <property name="text">
         <string>Available Digital Event Series:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTableWidget" name="audio_sources_table">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>120</height>
         </size>
        </property>
        <property name="alternatingRowColors">
         <bool>true</bool>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
        </property>
        <property name="gridStyle">
         <enum>Qt::PenStyle::SolidLine</enum>
        </property>
        <column>
         <property name="text">
          <string>Enabled</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Event Series Key</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Time Frame</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Event Count</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Volume</string>
         </property>
        </column>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
